book/src/getting-started/1-qobjects-in-rust.md

```diff
@@ -0,0 +1,77 @@
+
+> Don't bring a knife to a gun fight.
+
+There are many bits of advice like that.
+With CXX-Qt, we aim to make it possible to use the right tool for each of the many jobs necessary to build a modern GUI application.
+
+So what is in our toolbox for a typical Qt application?
+- QML - A declarative, flexible, dynamically-typed, interpreted language that is purpose built to define reactive and beautiful GUI layouts and widgets with quick iteration speed.
+- C++ - The traditional back-end of Qt - A fast, low-level language with a strong type system. C++ offers a rich ecosystem, many Qt-specific libraries and bare-metal performance. The cost when using C++ is that it is slow to develop, very error-prone and can easily lead to memory-issues, which can instantly crash your application and cause security issues.
+
+Notably absent then is a back-end language that allows us to get rid of the issues C++ has and provides us with a safe way to write fast Back-end code.
+This of course is where Rust comes in.
+Whilst Rust doesn't have quite as rich of an ecosystem, it is typically faster to develop than C++, with easy dependency management, and most importantly, safe memory access.
+Therefore it is an ideal candidate to replace C++ for writing the back-end business-logic code that feeds the GUI with data.
+
+However, both C++ as well as QML still have their place in Qt applications.
+For that reason Rust, C++ and QML should all be able to be used to complement each other.
+CXX-Qt aims to make it easy to integrate all three languages easily with each other, through the use of Qts [meta object system](https://doc.qt.io/qt-5/metaobjects.html).
+
+Qts design is inherently object-oriented, which is true both for C++ and QML as well.
+Therefore, in order to integrate well with Qt, Rust needs to be able to extend the Qt object system with its own QObject subclasses and instances.
+This is exactly what CXX-Qt allows you to do.
+
+As Rust doesn't offer classes with inheritance and polymorphism, CXX-Qt uses a Rust module when defining new QObject subclasses.
+
+These CXX-Qt modules consist of multiple parts:
+- A `Data` struct
+    - Defines which Properties will be in the QObject subclass.
+    - Needs to implement the `Default` trait.
+    - This data will live in a C++ subclass that is generated from the CXX-Qt module
+- A `RustObj` struct
+    - A normal Rust struct
+    - One struct instance is created per class instance
+    - Contains any Rust-only data
+- The `impl` of the `RustObj` struct (optional):
+    - Contains any Rust code
+    - Functions marked with `#[invokable]` will be callable from QML and C++
+
+CXX-Qt will then expand this Rust module into two separate parts:
+1. A C++ subclass of QObject with the same name as the module
+2. The Rust struct `RustObj`
+
+<div style="padding: 1rem; text-align: center;">
+
+![Overview of CXX-Qt module generation](../images/cxx_qt_module_generation.png)
```

Comment:
I'd prefer not to have binary images without their source in the code (in case it needs to be adjusted later). I can recreate this image in something like Dia with a SVG output if that's useful ?

Reply:
Also the file is missing it's equivalent `.license` file with licensing information (hence why reuse is failing in CI).

Reply:
A Dia file would be good :+1:

Reply:
As discussed this might reuse the existing one.

Reply:
Changed it to reuse the existing diagram